Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -127,6 +127,42 @@ void OpenALStream::Clear(bool mute)
   }
 }
 
+static ALenum CheckALError(const char* desc)
+{
+  ALenum err = alGetError();
+
+  if (err != AL_NO_ERROR)
+  {
+    std::string type;
+
+    switch (err)
```

Comment:
I'm personally a fan of using maps for things like that (because thats what you do here, map a value of type `ALenum` to `std::string`). Although, I do not know the appropriate container in C++ (`map`? `unordered_map`? Something else since both add a new value when using `operator[]` on a non-existing element?) for that; nor the appropriate pattern (similar to a `Dictionary<TKey, TValue>.TryGetValue(someTKey, out someTValue)` in C#)...so unless someone else speaks up, feel free to ignore this comment.